_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,539 @@
+---
+layout: post
+title:  "Reassigning partitions in Apache Kafka Cluster"
+date: 2021-09-23
+author: shubham_rawat
+---
+
+Apache Kafka brokers host replicas of partitions. How many and which partitions each broker replicates affects the load on that broker.
+One broker often ends up experiencing more load than others, and the most loaded broker can become a bottleneck within the cluster.
+Strimzi supports the use of Cruise Control and we recommend using it to balance the load across the brokers.
+That's because when there are more than a handful of partitions in your cluster it becomes very hard to find the best, or even a "good enough", assignment.
+But sometimes you might need to make small changes to the replica assignment.
+
+This blog post will guide you through when you might need to do reassignments using Kafka's partition reassignment rool, and how to do so.
+
+<!--more-->
+
+##  When to use the Kafka reassignment partition tool
+
+As mentioned, we recommend using Cruise Control where possible.
+Currently Strimzi offers first-class support for three use cases:
+
+* Global cluster balancing.
+  This uses the `KafkaRebalance` custom resource.
+* Automatically assigning replicas to newly-added brokers. This uses a `KafkaRebalance` with `spec.mode` set to `add-broker`, which was added in Strimzi 0.29.
+* Automatically reassigning replicas away from brokers prior to scaling down the cluster. This uses a `KafkaRebalance` with `spec.model` set to `remove-broker`. This was also added in Strimzi 0.29.
+
+These use cases are covered extensively in [the documentation](https://strimzi.io/docs/operators/latest/configuring.html#cruise-control-concepts-str).
+So if this is the kind of rebalancing you need you can stop reading the rest of this post.
+
+The Strimzi community is also planning further integration with Cruise Control:
+
+* Simplifying the existing rebalancing functionality using [auto-approval](https://github.com/strimzi/proposals/blob/main/038-optimization-proposal-autoapproval.md).
+* Better supporting [changing the replication factor](https://github.com/strimzi/proposals/pull/50) of individual topics using the Topic Operator.
+
+Until those improvements are implemented it's possible to use the reassignment tool for these cases.
```

Comment:
How auto-approval proposal/feature is related to the usage of the reassignment tool? I don't see the relationship. So I don't understand this sentence.